DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/component_interfaces_comparison/streaming-invocation/src/vector_add.cpp

```diff
@@ -0,0 +1,116 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/sycl.hpp>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class IDSimpleVAdd;
+
+struct SimpleVAddKernel {
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::intel::experimental::conduit})>
+      a_in;
+
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::intel::experimental::conduit})>
+      b_in;
+
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int *, decltype(sycl::ext::oneapi::experimental::properties{
+                 sycl::ext::intel::experimental::conduit})>
+      c_out;
+
+  sycl::ext::oneapi::experimental::annotated_arg<
+      int, decltype(sycl::ext::oneapi::experimental::properties{
+               sycl::ext::intel::experimental::conduit})>
+      len;
+
+  // kernel property method to config invocation interface
+  auto get(sycl::ext::oneapi::experimental::properties_tag) {
+    return sycl::ext::oneapi::experimental::properties{
+        sycl::ext::intel::experimental::streaming_interface<>,
+        sycl::ext::intel::experimental::pipelined<1>};
+  }
+
+  void operator()() const {
+    for (int idx = 0; idx < len; idx++) {
+      int a_val = a_in[idx];
+      int b_val = b_in[idx];
+      int sum = a_val + b_val;
+      c_out[idx] = sum;
+    }
+  }
+};
+
+#define VECT_SIZE 256
```

Comment:
Should avoid usage of #define in C++ and try to use a typesafe alternative like constexpr